src/tests/fileIOSuite.c

```diff
@@ -0,0 +1,99 @@
+#include <stdio.h>
+#include <stdlib.h>
+
+#ifdef _WIN32
+#include <fcntl.h>
+#include <io.h>
+#endif
+
+#include <string.h>
+
+#include "CuTest.h"
+#include "assertExtensions.h"
+#include "fileIO.h"
+
+#define TEST_STRING "Lorem ipsum dolor sit amet, consectetur adipiscing elit.\nMorbi dignissim nunc vel lectus egestas, nec rutrum nibh gravida.\n\t(){}[];:<>\\\"'"
+
+/* Test to check that contents of a file are the same as a string read from the same file */
+void test_read_write_roundtrip(CuTest *tc) {
+    const char *string = TEST_STRING;
+    char *read_string;
+
+    write_file("src/tests/testString.tmp", string);
+    read_string = read_file("src/tests/testString.tmp");
+
+    CuAssertStrEquals(tc, string, read_string);
+
+    free(read_string);
+}
+
+/* Test to check that a file is created on the specified path */
+void test_write_file_exists(CuTest *tc) {
+    const char *string = TEST_STRING;
+    FILE *filePtr;
+
+    write_file("src/tests/testString2.tmp", string);
+    CuAssertTrue(tc, (filePtr = fopen("src/tests/testString2.tmp", "r")) != NULL);
+
+    fclose(filePtr);
+}

```

Comment:
What if the file already exists? Then this would not test anything

Reply:
well no, but actually yes